models/article.go

```diff
@@ -0,0 +1,209 @@
+package models
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Machiel/slugify"
+	"github.com/jinzhu/gorm"
+)
+
+type ScopeHandler func(db *gorm.DB) *gorm.DB
+
+type ArticleStorer interface {
+	CreateArticle(*Article) error
+	DeleteArticle(*Article) error
+	GetAllArticles() *gorm.DB
+	GetAllArticlesAuthoredBy(string) ([]Article, error)
+	GetAllArticlesFavoritedBy(string) ([]Article, error)
+	GetAllArticlesWithTag(string) ([]Article, error)
+	GetArticle(string) (*Article, error)
+	FavoriteArticle(int, int) error
+	UnfavoriteArticle(int, int) error
+	FindUserByUsername(string) (*User, error)
+	IsFavorited(int, int) bool
+	IsFollowing(int, int) bool
+	SaveArticle(*Article) error
+}
+
+// Article article model
+type Article struct {
+	ID             int
+	Slug           string
+	Title          string
+	Description    string
+	Body           string
+	User           User
+	UserID         int
+	Tags           []Tag `gorm:"many2many:taggings;"`
+	Favorites      []Favorite
+	FavoritesCount int
+	CreatedAt      time.Time
+	UpdatedAt      time.Time
+}
+
+type ValidationMessages map[string]interface{}
+
+// NewArticle returns a new Article instance.
+func NewArticle(title string, description string, body string, user *User) *Article {
+	return &Article{
+		Title:       title,
+		Description: description,
+		Body:        body,
+		User:        *user,
+	}
+}
+
+// IsValid check if the article has a valid title, description and body
+func (a *Article) IsValid() (bool, map[string]interface{}) {
+	var errs = ValidationMessages{}
+	var valid = true
+
+	if a.Title == "" {
+		errs["title"] = []string{"title field can't be blank"}
+		valid = false
+	}
+
+	if a.Description == "" {
+		errs["description"] = []string{"description field can't be blank"}
+		valid = false
+	}
+
+	if a.Body == "" {
+		errs["body"] = []string{"body field can't be blank"}
+		valid = false
+	}
+
+	return valid, errs
+}
+
+func (a *Article) IsOwnedBy(username string) bool {
+	return a.User.Username == username
+}
+
+// CreateArticle persist a new article
+func (db *DB) CreateArticle(article *Article) (err error) {
+	err = db.Create(&article).Error
```

Comment:
Yeah found it on the so call documentation of mainly built with code snippets, I will not blame the author, I'm not a good at writing documentation so...